clients/cpp-console/src/common/DialogManager.cpp

```diff
@@ -407,4 +426,14 @@ void DialogManager::InitializeConnection()
     auto stringFuture = _dialogServiceConnector->SendActivityAsync(keywordPrimingActivityText);
 
     log_t("Connector successfully initialized!");
+}
+
+// Individual devices will have differing capabilities and interfaces for sharing
+// interaction state in a headless environment. This one implementation under a
+// general abstraction.
+void DialogManager::SetDeviceStatus(const DeviceStatus status)
+{
+    const std::string name_map[]{ "Idle", "Initializing", "Ready", "Detecting", "Listening", "Thinking", "Speaking" };
+
+    std::cout << "New status : " << name_map[(int)status] << std::endl;

```

Comment:
Find a way to log the current device state every second in the main file... maybe modify the while loop that is listening for key stroke to log every second. That would be a good way to test that the device state is correct. For example - when I start hearing the TTS audio playback, I see on screen that the device state has change to Speaking.
Since you don't want to write a new line every second, override the current line.

Reply:
I realize now that the reason the device status was in a separate file was for the GGEC speaker to use it differently, but still be able to share the main code. I suggest we have the state machine implemented as an object or interface that this object can inherit from or have that state machine as an object in the DialogManager that is used to keep track of the state. That way we can still share the code, but the State will be kept properly.

Reply:
In DialogManager class add 1. a private field _deviceStatus, 2. a private function SetDeviceStatus, 3.a public function GetDeviceStatus. The new log way is still under research.